doc/kubernetes/modules/ROOT/pages/running/metrics/jgrp_metrics.adoc

```diff
@@ -0,0 +1,125 @@
+
+TIP: In a healthy cluster, the thread pool should never be closer to its maximum size (by default, `200` threads).
+
+NOTE: Thread pool metrics are not available with virtual threads.
+
+|===
+|Metric |Description
+
+|`vendor_jgroups_tcp_get_thread_pool_size`
+|Current number of threads in the thread pool, for the intra-cluster communication.
+
+|`vendor_jgroups_tunnel_get_thread_pool_size`
+|Current number of threads in the thread pool, for the cross-site communication.
+
+|===
+
+== Flow Control
+
+Flow control takes care of adjusting the rate of a message sender to the rate of the slowest receiver over time.
+This is implemented through a credit based system, where each sender credits and decrements them whenever a message is sent.
+The sender blocks when the credits fall below 0, and only resumes sending messages when it receives a replenishment message from the receivers.
+
+The metrics below show the number of blocked messages and the average blocking time.
+When a value different from zero, it may signal that a receiver is overloaded and may degrade the cluster performance.
+
+Each node has two independent flow control protocols, `UFC` for unicast messages and `MFC` for multicast messages.
+Unicast messges are one-to-one messages while multicast are one-to-many messages.
+{ispn} mostly uses unicast messages.
+
+TIP: A healthy cluster shows a value of zero for all metrics.
+
+|===
+|Metric |Description
+
+| `vendor_jgroups_ufc_get_number_of_blockings`
+| The number of times flow control blocks sender for unicast messages.
+
+| `vendor_jgroups_ufc_get_average_time_blocked`
+| Average time blocked (in ms) in flow control when trying to send an unicast message.
+
+| `vendor_jgroups_mfc_get_number_of_blockings`
+| The number of times flow control blocks sender for multicast messages.
+
+| `vendor_jgroups_mfc_get_average_time_blocked`
+| Average time blocked (in ms) in flow control when trying to send a multicast message.
+
+|===
+
+== Retransmissions
+
+JGroups provides reliable delivery of messages.
+When a message is dropped, on the network or the receiver cannot handle the message, a retransmission is required.
+Retransmissions increase resources usage, and it is usually a signal of an overload system.
+
+Random Early Drop (RED) monitors the sender queues.
+When the queues are almost full, the message is dropped, and a retransmission must happen.
+It prevents threads from being blocked by a full sender queue.
+
+TIP: A healthy cluster shows a value of zero for all metrics.
+
+|===
+|Metric |Description
+
+| `vendor_jgroups_unicast3_get_num_xmits`
+| The number of retransmitted messages.
+
+| `vendor_jgroups_red_get_dropped_messages`
+| The total number of dropped message by the sender.
+
+| `vendor_jgroups_red_get_drop_rate`
+| Percentage of all messages that were dropped by the sender.
+
+|===
+
+== Network Partitions
+
+Network partitions in a cluster can happen due to various reason.
+The merge events and network partitions merging.
```

Comment:
I'm not sure the intent of this statement, I think it needs to be expanded.

Reply:
Late night writing:rofl: